practice/rain.py

```diff
@@ -2,129 +2,135 @@
+    """
+    >>> get_year_from_date_amount(('10-DEC-1988', 1))
+    '1988'
+    """
+    year_key = date_amount[0][-4:]
+    return year_key
+
+
+def get_year_with_most_rain(parsed_lines):
+    """Returns year with max amount of rain.
+
+    >>> get_year_with_most_rain([('10-DEC-1988', 1), ('14-JAN-1999', 10), ('01-OCT-1988', 4), ('05-MAY-1999', 3)])
+    '1999'
+    """
+    total_by_year_list = create_year_list(parsed_lines)
+    total_by_year_dict = create_year_dict(total_by_year_list)
+
+    total_by_year_max = max(total_by_year_dict, key=total_by_year_dict.get)
+    return total_by_year_max
+
+
+def create_year_list(parsed_lines):
     """Slices the list of dates into years
 
-    >>> create_year_list([['Dec1999', 'Party over! OOPS! out of time']])
+    >>> create_year_list([['DEC-1999', 'Party over! OOPS! out of time']])
     [['1999', 'Party over! OOPS! out of time']]
     """
-    temp_list3 = []
-    for date, rain in temp_list2:
+    list_of_years_and_rainfall = []
+    for date, rain in parsed_lines:
         date_sliced = date[-4:]
-        temp_list3 += [[date_sliced, rain]]
-    return temp_list3
+        list_of_years_and_rainfall += [[date_sliced, rain]]
+    return list_of_years_and_rainfall
 
 
-def create_year_dict(temp_list3):
+def create_year_dict(list_of_year_and_rainfall):
     """Creates a dict from year list and sums rain value.
 
     >>> create_year_dict([['1999', 1], ['1999', 2]])
     {'1999': [3]}
     """
     yearly_totals = {}
-    for date_sliced, rain in temp_list3:
+    for date_sliced, rain in list_of_year_and_rainfall:
         if date_sliced not in yearly_totals:
             yearly_totals[date_sliced] = []
         yearly_totals[date_sliced] += [rain]
-        total_by_year = dict(zip(yearly_totals.keys(), [[sum(item)] for item in yearly_totals.values()]))
+    total_by_year = dict(zip(yearly_totals.keys(), [[sum(item)] for item in yearly_totals.values()]))
     return total_by_year
 
 
-def get_max_year(total_by_year):
-    """Returns year with max amount of rain.
+def output(day_with_most_rain, total_by_year_max):
+    """Combines output strings for program.
 
-    >>> get_max_year({'1999':200, '1998':100})
-    '1999'
+    >>> output('DEC_31-1999', '1922')
+    The date that had the most rain was: DEC_31-1999
+    1922 was the wettest year from this data sample.
     """
-    total_by_year_max = max(total_by_year, key=total_by_year.get)
-    return total_by_year_max
-
-
-def output(day_with_most_rain, total_by_year_max):
-    """Combines output strings for program."""
-    output_one = 'The date that had the most rain was: ' + str(day_with_most_rain)
-    output_two = str(total_by_year_max) + ' was the wettest year from this data sample.'
-    output_three = str(output_one) + '\n' + output_two
+    output_one = 'The date that had the most rain was: ' + str.format(day_with_most_rain)
+    output_two = str.format(total_by_year_max) + ' was the wettest year from this data sample.'
```

Comment:
Not sure I understand what str.format or format() is doing. Output looks the same if I use them or not, I used str.format because you previously asked me to use it. I also do not know what the {} are doing in your example.